src/box/errcode.h

```diff
@@ -395,6 +395,8 @@ struct errcode_record {
 	_(ER_IN_ANOTHER_PROMOTE, 278,		"box.ctl.promote() is already running") \
 	_(ER_SHUTDOWN, 279,			"Server is shutting down") \
 	_(ER_FIELD_VALUE_OUT_OF_RANGE, 280,	"The value of field %s exceeds the supported range for type '%s': expected [%s..%s], got %s") \
+	_(ER_PRIV_NOT_GRANTED_BY_ORIGIN, 281,	"%s '%s' does not have %s access on %s '%s' provided by %s origin") \
+	_(ER_ROLE_NOT_GRANTED_BY_ORIGIN, 282,	"%s '%s' does not have role '%s' provided by %s origin") \
```

Comment:
Please use payload (see `TEST_ERROR_CODES` and/or ask @nshy for an example).

Reply:
AFAIR, it will not be shown (almost) anywhere while the `box_error_serialize_verbose` compat option remains `old`. But IMHO the old message is just incorrect in the case. It states that there are no privilege, when it is present (just granted by another origin).

Reply:
In the new version we use a special `box.error()` syntax that allows to use a known error code but a different error message.